### PR TITLE
CI Cache

### PR DESCRIPTION
**Pull request by q349wang**
Refactors CI to use ccache for compilation results

Ripped out windows builds REST IN PIECES BOZO

CHECK OUT THIS SICK IMPACT
BEFORE:
image
 ^ ABOSULTE SLOG FEST
 
 AFTER:
 image
^ MF AINT REBUILDING SHIT WOOOOOOOOOO

### CODE REVIEW

**Comment by q349wang** [comment]
Credit: https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/

File: CMakeLists.txt [hunk 1]
```diff
@@ -134,8 +149,14 @@ target_include_directories(antioch-base-test PUBLIC
 target_link_libraries(antioch-base-test PUBLIC
   antioch-base
   GTest::gtest_main
-  GTest::gmock_main
 )
+include(GoogleTest)
+# Windows environment on GitHub Actions is quirky
+if (DEFINED ENV{GITHUB_ACTIONS} AND WIN32)
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt]: don’t need this anymore

File: CMakeLists.txt [hunk 2]
```diff
@@ -28,6 +28,15 @@ endif()
 
 add_compile_options(-fno-rtti)
 
+enable_testing()
+FetchContent_Declare(
```
Comment (Victor-C-Zhang) [inline CMakeLists.txt:32]: this dupes the fetchcontent a few lines down. one of them’s gotta go

**Review by Victor-C-Zhang — APPROVED** [review]
some small things

**Pull request merged**